keras_nlp/layers/random_deletion_test.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for Transformer Decoder."""
+
+import tensorflow as tf
+
+from keras_nlp.layers import random_deletion
+
+
+class RandomDeletionTest(tf.test.TestCase):
```

Comment:
We need to add more test coverage here. Particularly

 - Calling the layer on an unbatched dataset using `tf.data.Dataset.map`.
 - Calling the layer on an batched dataset using `tf.data.Dataset.map`.
 - Compiling the layer into a simple keras model that takes a string input and outputs a string with deletions.

This should help us uncover where we need to improve the call graph code in the main class.

Reply:
Still need to do this!